src/app/settings/keyboard-mapping-settings/components/edit-keyboard-mapping/edit-keyboard-mapping.component.ts

```diff
@@ -97,4 +103,38 @@ export class EditKeyboardMappingComponent implements OnInit {
   public cancel(): void {
     this.onCancel.emit()
   }
+
+  public get saveButtonErrorsTooltip(): string | undefined {
+    const invalidControlsLabels: string[] = this.findInvalidControlsLabels()
+    if (invalidControlsLabels.length > 0) {
+      return $localize`common.form-required-field.button-tooltip` + `: ${invalidControlsLabels.join(', ')}`
+    } else {
+      return undefined
+    }
+  }
+
+  private findInvalidControlsLabels(): string[] {
+    return Object.entries({ ...this.actionTriggerForm.controls, ...this.actionTriggerDataForm.controls })
+      .filter(([name, control]) => !control.valid && name !== KEYBOARD_DATA_CONTROL_ID)
+      .map(([name]) => this.translateControl(name))
+  }
+
+  private translateControl(controlName: string): string {
+    switch (controlName) {
+      case KEYBOARD_TRIGGER_ON_CONTROL_ID:
```

Comment:
There is still a semantic mismatch between control name and control id. Only one term should be used :)

Reply:
Fixed